llap-server/src/java/org/apache/hadoop/hive/llap/cache/BuddyAllocator.java

```diff
@@ -294,25 +385,18 @@ public void allocateMultiple(MemoryBuffer[] dest, int size, BufferObjectFactory
     // allocator thread (or threads per arena).
     // The 2nd one is probably much simpler and will allow us to get rid of a lot of sync code.
     // But for now we will just retry. We will evict more each time.
-    int attempt = 0;
+    int discardsAttempt = 0;
     boolean isFailed = false;
     int memoryForceReleased = 0;
     try {
       int discardFailed = 0;
       while (true) {
         // Try to split bigger blocks.
-        int startArenaIx = (int)((threadId + attempt) % arenaCount);
+        startArenaIx = (int)((threadId + discardsAttempt) % arenaCount);
```

Comment:
Maybe we want to add forceEvictAttempt here, not to start at the first (threadID given) arena again?

Reply:
done